config/config.go

```diff
@@ -365,8 +369,13 @@ func (c *Config) UnmarshalYAML(unmarshal func(interface{}) error) error {
 			if ec.AuthUsername == "" {
 				ec.AuthUsername = c.Global.SMTPAuthUsername
 			}
-			if ec.AuthPassword == "" {
+			// require a password only if a username is provided
```

Comment:
I don't see how this change relates to the original issue? Can you elaborate?

Reply:
Another good catch.  I thought username-but-no-password was an invalid config, but now see that it's [explicitly supported](https://github.com/prometheus/alertmanager/blob/v0.24.0/notify/email/email.go#L77).  Simplified.

Reply:
Ugh, wait, sorry, I was rushing, and misread the code in that last part.  no-username is supported, but username-without-password still produces errors.  Hold on, let me get this sorted.  Sorry for the hassle.

Reply:
Okay, I finally remembered. Local configs inherit fields from the global config, and now that the SMTP password can come from two different kinds of field -- literal or from-file -- I wanted to avoid having a local config with one kind of field inheriting the other kind of field from global. I.e. if the global config has SMTPAuthPasswordFile set, and a local config has AuthPassword set, that local config should not inherit SMTPAuthPasswordFile.

I updated the code again to enforce that; local configs now only inherit the global SMTP password fields if both of the local fields are unset.